crates/ruma-identifiers/src/device_id.rs

```diff
@@ -6,6 +6,21 @@ opaque_identifier! {
     ///
     /// Device identifiers in Matrix are completely opaque character sequences. This type is
     /// provided simply for its semantic value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ruma_identifiers::DeviceId;
+    ///
+    /// let random_id = DeviceId::new();
+    /// assert_eq!(random_id.as_str().len(), 8);
+    ///
+    /// let ref_id: &DeviceId = "abcdefghi".into();
```

Comment:
A different way to do the same thing is `let ref_id = device_id!("abcdefghi");`. Maybe add that as `let static_id`?

Reply:
> Sorry that that got you stuck on doing something so simple.

Ruma is awesome, don't be sorry!  I am working on this change, thank you.